File: USACO/Silver/Python/USACO 2019 Feb Silver Painting the Barn.py
import sys

sys.stdin = open('paintbarn.in', 'r')
sys.stdout = open('paintbarn.out', 'w')

input = sys.stdin.readline

n, k = map(int, input().split())

dp = [[0]*1001 for _ in range(1001)]
for _ in range(n):
    x1, y1, x2, y2 = map(int, input().split())
    dp[x1][y1] += 1
    dp[x1][y2] -= 1
    dp[x2][y1] -= 1
    dp[x2][y2] += 1

ans = 0
for i in range(1001):
    for j in range(1001):
        if i > 0:
            dp[i][j] += dp[i-1][j]
        if j > 0:
            dp[i][j] += dp[i][j-1]
        if i > 0 and j > 0:
            dp[i][j] -= dp[i-1][j-1]
        if dp[i][j] == k:
            ans += 1

print(ans)

File: USACO/Silver/C++/USACO 2019 Feb Silver Painting the Barn.cpp
#include <bits/stdc++.h>
using namespace std;

#define int long long
#define MOD(a, b) ((((a) % (b)) + (b)) % (b))

bool cbs(pair<int, int> a, pair<int, int> b) {
    return a.second < b.second;
}
const int MAXN = 1001;

signed main() {
    freopen("paintbarn.in", "r", stdin);
    freopen("paintbarn.out", "w", stdout);

    cin.tie(NULL) -> sync_with_stdio(false);
    
    int n, k;
    cin >> n >> k;
    vector<vector<int>> grid(MAXN, vector<int>(MAXN));
    while(n--){
        int x1, y1, x2, y2;
        cin >> x1 >> y1 >> x2 >> y2;
        ++grid[x1][y1];
        --grid[x2][y1];
        --grid[x1][y2];
        ++grid[x2][y2];
    }
    int ans=0;
    for(int i=0; i<1000; ++i){
        for(int j=0; j<1000; ++j){
            if (i) grid[i][j] += grid[i-1][j];
            if (j) grid[i][j] += grid[i][j-1];
            if (i && j)grid[i][j] -= grid[i-1][j-1];
            if(grid[i][j] == k) ++ans;
        }
    }
    cout << ans;
}
